src/components/AddTodo.jsx

```diff
@@ -25,40 +25,50 @@ const AddTodo = (props = (todos, setTodos)) => {
         }));
     };
 
-    const handleCreateTodo = () => {
+    const handleCreateTodo = (event) => {
         axios
             .post(import.meta.env.VITE_TODO_API, data)
-            .then((res) => props.setTodos([...props.todos, res.data]))
+            .then((res) => setTodos([...todos, res.data]))
             .then(navigateTo("/"));
+        event.preventDefault();
```

Comment:
this line should be the first thing you execute in the submit function, so put it before the axios call